internal/pkg/cli/pipeline_init.go

```diff
@@ -150,11 +150,25 @@ func newInitPipelineOpts(vars initPipelineVars) (*initPipelineOpts, error) {
 
 // Validate returns an error if the flag values passed by the user are invalid.
 func (o *initPipelineOpts) Validate() error {
-	// TODO add validation for flags
-	if o.appName == "" {
-		return errNoAppInWorkspace
-	}
+	// May consider validating githubAccessToken and gitBranch flag values in the future.
 
+	if o.URL != "" {
+		if err := validateDomainName(o.URL); err != nil {
+			return err
+		}
+		if !strings.Contains(o.URL, githubURL) && !strings.Contains(o.URL, "codecommit") {
```

Comment:
```suggestion
		if !strings.Contains(o.URL, githubURL) && !strings.Contains(o.URL, codeCommitURL) {
```

Can we extract this to a const like we do with `githubURL`?

Reply:
Ah yes. I had done that in my larger CodeCommit branch but hadn't carried that over to this one. Will do, thanks!

Reply:
Yup. It's now "ccSubstring" bc CodeCommit doesn't use the URL like GH does.